File: backend/src/domain/entities/receipt.go
package entities

import "time"

type Receipt struct {
	ID                string    `json:"id" bson:"_id,omitempty"`
	ShopID            string    `json:"shop_id" bson:"shop_id"`
	PaymentMethod     string    `json:"payment_method" bson:"payment_method"` // e.g., "cash"
	TotalAmount       float64   `json:"total_amount" bson:"total_amount"`
	VatRate           float64   `json:"vat_rate" bson:"vat_rate"` // e.g. 0.07
	VAT               float64   `json:"vat" bson:"vat"`
	NetTotal          float64   `json:"net_total" bson:"net_total"` // ยอดสุทธิ (Total + VAT)
	Status            string    `json:"status" bson:"status"`       // เช่น "completed", "cancelled"
	CustomerRequestID string    `json:"customer_request_id,omitempty" bson:"customer_request_id,omitempty"`
	CreatedAt         time.Time `json:"created_at" bson:"created_at"`
}

type ReceiptItem struct {
	ID        string  `json:"id" bson:"_id,omitempty"`
	ReceiptID string  `json:"receipt_id" bson:"receipt_id"`
	ShopID    string  `json:"shop_id" bson:"shop_id"`
	WasteID   string  `json:"waste_id" bson:"waste_id"`
	Name      string  `json:"name" bson:"name"`             // Snapshot of name
	Category  string  `json:"category" bson:"category"`     // Snapshot of category
	Weight    float64 `json:"weight" bson:"weight"`         // kg
	UnitPrice float64 `json:"unit_price" bson:"unit_price"` // Price per unit at that time
	Price     float64 `json:"price" bson:"price"`           // Total price for this item
}

File: frontend/src/types/receipt.ts
export interface Receipt {
  id: string
  shop_id: string
  payment_method: string
  total_amount: number
  vat_rate: number
  vat: number
  net_total: number
  status: string
  customer_request_id?: string
  created_at: string
}

export interface ReceiptItem {
  id: string
  receipt_id: string
  shop_id: string
  waste_id: string
  name: string
  category: string
  weight: number
  unit_price: number
  price: number
}

export interface MonthlyCategoryStat {
  category: string
  weight: number
}

export interface DailyStat {
  date: string
  weight: number
}

export interface UserAnalyticsResponse {
  total_weight_this_month: number
  total_weight_last_month: number
  weight_diff_percent: number
  total_earnings_this_month: number
  monthly_category_stats: MonthlyCategoryStat[]
  daily_stats: DailyStat[]
}
